### PR TITLE
Add support for multiple tags

### PR DESCRIPTION
**Pull request by maciejzgadzaj**

### CODE REVIEW

File: lib/openapi3.js [hunk 1]
```diff
@@ -44,18 +44,18 @@ function convertToToc(source,data) {
                     sMethodUniqueName = method.operation.summary;
                 }
                 method.slug = sMethodUniqueName.toLowerCase().split(' ').join('-'); // TODO {, } and : ?
-                var tagName = data.translations.defaultTag;
-                var tagDescription = '';
-                if (method.operation.tags && method.operation.tags.length > 0) {
-                    var tagData = getTagGroup(method.operation.tags[0], data.options.tagGroups);
-                    tagName = tagData.name;
-                    tagDescription = tagData.description;
+                if (typeof method.operation.tags === 'undefined' || !method.operation.tags.length) {
```
Comment (IdoO2) [inline lib/openapi3.js:47]: `method.operation.tags` was left undefined in the original code. Meaning that you are not only allowing a method to be attached to several tags, but also making all methods associated with at least one tag, the default?

File: lib/openapi3.js [hunk 2]
```diff
@@ -44,18 +44,18 @@ function convertToToc(source,data) {
...
+                    method.operation.tags = [data.translations.defaultTag];
                 }
-                if (!resources[tagName]) {
-                    resources[tagName] = { count: 0, methods: {}, description: tagDescription};
+                for (let t in method.operation.tags) {
```
Comment (IdoO2) [inline lib/openapi3.js]: If I read correctly, `method.operation.tags` is an array, so `method.operation.tags.forEach(...)` would make more sense.

File: lib/openapi3.js [hunk 3]
```diff
@@ -44,18 +44,18 @@ function convertToToc(source,data) {
...
+                    let tagData = getTagGroup(method.operation.tags[t], data.options.tagGroups);
```
Comment (IdoO2) [inline lib/openapi3.js]:
```suggestion
                    const tagData = getTagGroup(method.operation.tags[t], data.options.tagGroups);
```

File: lib/openapi3.js [hunk 4]
```diff
@@ -44,18 +44,18 @@ function convertToToc(source,data) {
...
+                    let tagName = tagData.name;
```
Comment (IdoO2) [inline lib/openapi3.js]:
```suggestion
                    const tagName = tagData.name;
```

File: lib/openapi3.js (hunk 1 above)
Comment (maciejzgadzaj) [reply lib/openapi3.js:47]: Yup, but I don't care. 😎  On a more serious note though, this code adds method to the resources with the defaultTag anyway if no tags were defined, so it actually makes sense to have the tag added to the method itself too. Not that it's used for anything later on. Which is to say, personally I'm fine with this behaviour change, but might still yield to maintainer's will.

Comment (IdoO2) [reply lib/openapi3.js:47]: Ok.

File: lib/openapi3.js [hunk 5]
```diff
@@ -44,18 +44,18 @@ function convertToToc(source,data) {
...
-                    tagDescription = tagData.description;
+                if (typeof method.operation.tags === 'undefined' || !method.operation.tags.length) {
+                    method.operation.tags = [data.translations.defaultTag];
                 }
-                if (!resources[tagName]) {
-                    resources[tagName] = { count: 0, methods: {}, description: tagDescription};
-                }
-                resources[tagName].count++;
-                resources[tagName].methods[sMethodUniqueName] = method;
+                method.operation.tags.forEach(function(tag){
```
Comment (IdoO2) [inline lib/openapi3.js:50]: Informational: from a purely syntactic standpoint, `function () {}` means `function <anonymous>()`; `function doSth() {}` means `function with name doSth`; `var doSth = function () {}` means `create anonymous function and assign it to var doSth`.  In the current case note that `function() {}` creates a conflict between anonymous and named (with reserved word `function`).

More to the point, if an error occurs, anonymous functions are a pain, so naming things helps (`forEach(function addTag(tag) {}`).

**Review by IdoO2 — APPROVED** [review]

**Pull request merged**